### ISSUE TITLE
[feature] Add shell completions

### ISSUE THREAD

**Issue by Freed-Wu** [mvdan/sh#914]
How about adding shell completions for bash/zsh/fish/posh/etc? Thanks.

```
❯ shfmt -<TAB>
option
--binary-next-line   binary ops like && and | may start a line
-bn                  binary ops like && and | may start a line
--case-indent        switch cases will be indented
-ci                  switch cases will be indented
...
```

**Comment by mvdan** [comment]
Sure, I don't see why not if someone wants to write and maintain that. Ideally there would be a way to do this automatically.

**Comment by Freed-Wu** [comment]
> Ideally there would be a way to do this automatically.

Like <https://jmtirado.net/shell-completion-with-cobra-and-go/>?

**Comment by mvdan** [comment]
Yeah, something like that. I want to avoid heavy dependencies like cobra, though. Aside from tests, we are currently light on dependencies, and I want to keep it that way.

**Comment by mvdan** [comment]
This has been open for over three years, so it seems like there's no interest in researching a solution to this. Closing it for now. Note that https://github.com/mvdan/sh/pull/1341 was raised, but it simply means maintaining scripts which duplicate all flags, which is added maintenance burden.

If someone wants AI-assisted completion, it should still be possible to consume the output of `shfmt --help` and generate from that yourselves.
